src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -906,7 +889,7 @@ bool AnalyzeGenericInstantiationTypeArray (in MessageOrigin origin, bool diagnos
 						// https://github.com/dotnet/linker/issues/2428
 						// We need to report the target as "this" - as that was the previous behavior
 						// but with the annotation from the generic parameter.
-						var targetValue = GetMethodParameterValue (calledMethod, 0, _context.Annotations.FlowAnnotations.GetGenericParameterAnnotation (genericParameters[i]));
+						var targetValue = GetMethodParameterValueInternal (calledMethod, 0, _context.Annotations.FlowAnnotations.GetGenericParameterAnnotation (genericParameters[i]));
```

Comment:
Why do we call the internal method here?

Reply:
I interpreted it as "internal to ReflectionMethodBodyScanner" but maybe that was meant more as a helper for the `GetMethodParameterValue` overload that I removed. I renamed it to just be an overload of `GetMethodParameterValue`.